extensions/default/src/DicomJSONDataSource/index.js

```diff
@@ -258,19 +262,19 @@ function createDicomJSONApi(dicomJsonConfig) {
 
       const { StudyInstanceUID, SeriesInstanceUID } = displaySet;
       const study = findStudies('StudyInstanceUID', StudyInstanceUID)[0];
-      const series = study.series.find(s => s.SeriesInstanceUID === SeriesInstanceUID);
-      let instances = displaySet.images;
-      if (series.instances.length > displaySet.images.length) {
-        instances = series.instances;
-      }
+      const series = study.series.find(s => s.SeriesInstanceUID === SeriesInstanceUID) || [];
 
-      instances.forEach(instance => {
+      displaySet.images.forEach(instance => {
         const NumberOfFrames = instance.NumberOfFrames;
-
         if (NumberOfFrames > 1) {
           for (let i = 0; i < NumberOfFrames; i++) {
+            /** In case there are multiple sop class we filter sop instances */
```

Comment:
This comment doesn't make sense - you are referring to sop class, but looking at sop instance uid.

Reply:
If you have a multiframe series with mixed sop class you might have different sop instance uid. I'm filtering here.

Reply:
If you have a multiframe series without mixed sop class you might have different sop instance uid - actually you WILL have different sop instance uid, one per instance object, so the whole thing is independent of sop class.